diff --git a/camlibs/ptp2/ptp.c b/camlibs/ptp2/ptp.c
index d0e5ace..c433080 100644
--- a/camlibs/ptp2/ptp.c
+++ b/camlibs/ptp2/ptp.c
@@ -1522,6 +1522,7 @@ ptp_panasonic_9415 (PTPParams* params, PanasonicLiveViewSize *liveviewsize)
 {
 	PTPContainer	ptp;
 	unsigned char	data[16] = {0};
+	unsigned char	*dataptr = data;
 
 	PTP_CNT_INIT(ptp, PTP_OC_PANASONIC_SetLiveViewParameters, 0x0d800011);
 
@@ -1532,7 +1533,7 @@ ptp_panasonic_9415 (PTPParams* params, PanasonicLiveViewSize *liveviewsize)
 	htod16a(data+12, liveviewsize->x);
 	htod16a(data+14, liveviewsize->freq);
 
-	return ptp_transaction(params, &ptp, PTP_DP_SENDDATA, sizeof(data), (unsigned char**)&data, 0);
+	return ptp_transaction(params, &ptp, PTP_DP_SENDDATA, sizeof(data), &dataptr, 0);
 }
 
 uint16_t
